circup/__init__.py

```diff
@@ -230,32 +237,53 @@ class Module:
     # pylint: disable=too-many-arguments
 
     def __init__(
-        self, path, repo, device_version, bundle_version, mpy, bundle, compatibility
+        self,
+        name,
+        backend,
+        repo,
+        device_version,
+        bundle_version,
+        mpy,
+        bundle,
+        compatibility,
     ):
         """
         The ``self.file`` and ``self.name`` attributes are constructed from
         the ``path`` value. If the path is to a directory based module, the
         resulting self.file value will be None, and the name will be the
         basename of the directory path.
 
-        :param str path: The path to the module on the connected
-            CIRCUITPYTHON device.
+        :param str name: The file name of the module.
+        :param Backend backend: The backend that the module is on.
         :param str repo: The URL of the Git repository for this module.
         :param str device_version: The semver value for the version on device.
         :param str bundle_version: The semver value for the version in bundle.
         :param bool mpy: Flag to indicate if the module is byte-code compiled.
         :param Bundle bundle: Bundle object where the module is located.
         :param (str,str) compatibility: Min and max versions of CP compatible with the mpy.
         """
-        self.path = path
-        if os.path.isfile(self.path):
-            # Single file module.
-            self.file = os.path.basename(path)
-            self.name = self.file.replace(".py", "").replace(".mpy", "")
+        self.name = name
+        self.backend = backend
+        self.path = os.path.join(backend.library_path, name)
+        url = urlparse(self.path)
+        if url.scheme == "http":
```

Comment:
This doesn't seem like the right place for this. What is the important state that is computed here? Why isn't the backend passing it in?